modules/@apostrophecms/schema/index.js

```diff
@@ -467,6 +467,27 @@ module.exports = {
       }
     });
 
+    self.addFieldType({
+      name: 'email',
+      vueComponent: 'AposInputString',
+      convert: function (req, field, data, object) {
+        object[field.name] = self.apos.launder.string(data[field.name], undefined, field.min, field.max);
+        if (!data[field.name].length) {
+          if (field.required) {
+            throw self.apos.error('required');
+          }
+        } else {
+          // regex source: https://emailregex.com/
+          // eslint-disable-next-line no-useless-escape
```

Comment:
This "useless-escapes" should be fixable rather than disabling the linter.

Reply:
I used the code from 2.x, event. I remember a conversation with Tom about it when I tried to adapt the regex to the linter. We agreed to keep the ignore rule instead of modifying the regex.

Reply:
As I recall this lint just plain wasn't right, the regexp did not cooperate.

Reply:
(When modified to please lint.)

Reply:
I don't think it's wrong. It's reacting to `[^<>()\[\]\\.,;:\s@"]`, which is excluding all the characters in that set. The closing bracket in the middle (`/]`) needs to be escaped to not end the set, but the opening bracket is inside the set (`[]`) so it's interpreted as the actual bracket character.

Changing it to `/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/` resolves the linter error and does the same thing.

Reply:
Done on front and back. Seems to work as expected during my tests.